verification/helpers.go

```diff
@@ -32,6 +68,21 @@ func getArtifactPathFromUri(artifactUri string) (string, error) {
 	return artifactPath, nil
 }
 
+func getArtifactDigestFromUri(artifactUri string) (string, error) {
+	invalidUriErr := fmt.Errorf("artifact URI %q could not be parsed, make sure it is the fully qualified OCI artifact URI without the scheme/protocol. e.g domain.com:80/my/repository@sha256:digest", artifactUri)
+	i := strings.LastIndex(artifactUri, "@")
+	if i < 0 || i+1 == len(artifactUri) {
+		return "", invalidUriErr
+	}
+
+	j := strings.LastIndex(artifactUri[i+1:], ":")
+	if j < 0 || j+1 == len(artifactUri[i+1:]) {
+		return "", invalidUriErr
+	}
+
+	return artifactUri[i+1:], nil
+}
```

Comment:
This block of code is workable but fragile. Try [registry.ParseReference()](https://pkg.go.dev/oras.land/oras-go/v2@v2.0.0-rc.1/registry#ParseReference).